runtime/caml/memory.h

```diff
@@ -412,16 +368,20 @@ struct caml__roots_block {
 
 #define CAMLxparamN(x, size) \
   struct caml__roots_block caml__roots_##x; \
-  CAMLunused_start int caml__dummy_##x = (     \
-    (void) caml__frame, \
-    (caml__roots_##x.next = Caml_state_field(local_roots)), \
-    (Caml_state_field(local_roots) = &caml__roots_##x), \
+  CAMLunused_start int caml__dummy_##x = ( \
+    (caml__roots_##x.next = *caml_local_roots_ptr), \
+    (*caml_local_roots_ptr = &caml__roots_##x), \
+    (caml__roots_##x.mutexes = 0), \
     (caml__roots_##x.nitems = (size)), \
     (caml__roots_##x.ntables = 1), \
     (caml__roots_##x.tables[0] = &(x[0])), \
     0) \
   CAMLunused_end
 
+#define CAMLxparamNextend(x, new_size) \
+  (caml__roots_##x.nitems = (new_size)) \
+  CAMLunused_end
```

Comment:
Is there any documentation for this macro? I'm very surprised to see an unbalanced `CAMLunused_end` here.

Reply:
I've not seen `CAMLxparamNextend` before, as far as I can tell it isn't used. 
With that in mind, I'm in favour of removal.

Reply:
This was me https://github.com/ocaml-multicore/ocaml-multicore/commit/1fb5f20e79f316996f11ef86d8d3924430eb7bda , it was a performance optimisation we needed in `caml_hash` for the concurrent GC that we no longer need. I will remove it.

Reply:
Fixed in d208873